contracts/bridge/contracts/binance/BasicInboundLane.sol

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.6.0 <0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "../interfaces/ICrossChainFilter.sol";
+import "./SubstrateMessageCommitment.sol";
+import "./SubstrateOutboundLane.sol";
+
+/**
+ * @title A entry contract for syncing message from Darwinia to Ethereum-like chain
+ * @author echo
+ * @notice The basic inbound lane is the message layer of the bridge
+ * @dev See https://itering.notion.site/Basic-Message-Channel-c41f0c9e453c478abb68e93f6a067c52
+ */
+contract BasicInboundLane is SubstrateMessageCommitment, SubstrateOutboundLane {
+    /**
+     * @notice Notifies an observer that the message has dispatched
+     * @param nonce The message nonce
+     * @param result The message result
+     * @param returndata The return data of message call, when return false, it's the reason of the error
+     */
+    event MessageDispatched(uint256 indexed lanePosition, uint256 indexed nonce, bool indexed result, bytes returndata);
+    event DeliveredMessagePruned(uint256 indexed lanePosition, uint256 indexed nonce);
+
+    /* Constants */
+
+    /**
+     * @dev Gas used per message needs to be less than 100000 wei
+     */
+    uint256 public constant MAX_GAS_PER_MESSAGE = 100000;
+
+    uint256 public constant MAX_UNCONFIRMED_MESSAGES = 50;
+    /**
+     * @dev Gas buffer for executing `submit` tx
+     */
+    uint256 public constant GAS_BUFFER = 60000;
+
+    struct DeliveredMessage {
+        address relayer;
+        bool dispatch_result;
+    }
```

Comment:
The trade off design for batch deliver and one-by-one?

Will it save storage space by using

```
struct DeliveredMessages {
        address relayer;
        uin256 begin;
        uin256 end;
        bytes dispatch_result_bit_vec;
}
```